src/sqlfluff/cli/commands.py

```diff
@@ -365,6 +374,26 @@ def lint(
         click.echo(json.dumps(result.as_records()))
     elif format == "yaml":
         click.echo(yaml.dump(result.as_records()))
+    elif format == "github-annotation":
+        github_result = []
+        for record in result.as_records():
+            filepath = record["filepath"]
+            for violation in record["violations"]:
+                # NOTE: The output format is designed for this GitHub action:
+                # https://github.com/yuzutech/annotations-action
+                # It is similar, but not identical, to the native GitHub format:
+                # https://docs.github.com/en/rest/reference/checks#annotations-items
+                github_result.append(
```

Comment:
This structure works for v2.0 of that github action. We ran into issues using version 3. 

We got further direction from that team on the issue we opened up with them: https://github.com/yuzutech/annotations-action/issues/36